util/deadlockhistory/deadlock_history.go

```diff
@@ -74,7 +76,7 @@ func NewDeadlockHistory(capacity int) *DeadlockHistory {
 // GlobalDeadlockHistory is the global instance of DeadlockHistory, which is used to maintain recent several recent
 // deadlock events globally.
 // TODO: Make the capacity configurable
-var GlobalDeadlockHistory = NewDeadlockHistory(10)
+var GlobalDeadlockHistory = NewDeadlockHistory(config.GetGlobalConfig().PessimisticTxn.DeadlockHistoryCapacity)
```

Comment:
I'm not sure about the design here, if the capacity is fixed during constructing, then does it means the user has to restart the tidb instance to enlarge it? Is it the preferred behaviour?

Reply:
It seems not correct to initialize the global variable directly with the global config, which I think it's not loaded at that time. I suggest you initialize it the first time it's used. (give a function like `GetGlobalDeadlockHistory` instead of accessing the global variable directly).

The collection's implementation is a simple deque-like thing, which I think you can implement a resize function for it. If you have difficulty in it, I can help you too.

For http api, it's in `/settings` path.  
https://github.com/pingcap/tidb/blob/348abf01f90eb19db09d72261cd83c37e563a813/server/http_status.go#L120
https://github.com/pingcap/tidb/blob/348abf01f90eb19db09d72261cd83c37e563a813/server/http_handler.go#L614